prog/test/vm_host.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+require "net/ssh"
+
+class Prog::Test::VmHost < Prog::Base
+  subject_is :sshable, :vm_host
+
+  label def start
+    hop_setup_vms
+  end
+
+  label def setup_vms
+    project = Project.create_with_id(name: "project 1", provider: "hetzner")
+    project.create_hyper_tag(project)
+
+    subnet1_s = Prog::Vnet::SubnetNexus.assemble(
+      project.id, name: "the-first-subnet", location: "hetzner-hel1"
+    )
+
+    subnet2_s = Prog::Vnet::SubnetNexus.assemble(
+      project.id, name: "the-second-subnet", location: "hetzner-hel1"
+    )
+
+    strand.add_child(subnet1_s)
+    strand.add_child(subnet2_s)
+
+    keypair_1 = SshKey.generate
+    keypair_2 = SshKey.generate
+    keypair_3 = SshKey.generate
+
+    vm1_s = Prog::Vm::Nexus.assemble(
+      keypair_1.public_key, project.id,
+      private_subnet_id: subnet1_s.id,
+      enable_ip4: true
+    )
+
+    vm2_s = Prog::Vm::Nexus.assemble(
+      keypair_2.public_key, project.id,
+      private_subnet_id: subnet1_s.id,
+      enable_ip4: true
+    )
+
+    vm3_s = Prog::Vm::Nexus.assemble(
+      keypair_3.public_key, project.id,
+      private_subnet_id: subnet2_s.id,
+      enable_ip4: true
+    )
+
+    Sshable.create(
+      unix_user: "ubi",
+      host: "temp_#{vm1_s.id}",
+      raw_private_key_1: keypair_1.keypair
+    ) { _1.id = vm1_s.id }
+
+    Sshable.create(
+      unix_user: "ubi",
+      host: "temp_#{vm2_s.id}",
+      raw_private_key_1: keypair_2.keypair
+    ) { _1.id = vm2_s.id }
+
+    Sshable.create(
+      unix_user: "ubi",
+      host: "temp_#{vm3_s.id}",
+      raw_private_key_1: keypair_3.keypair
+    ) { _1.id = vm3_s.id }
+
+    strand.add_child(vm1_s)
+    strand.add_child(vm2_s)
+    strand.add_child(vm3_s)
+    strand.add_child(Strand[vm1_s.vm.nics.first.id])
+    strand.add_child(Strand[vm2_s.vm.nics.first.id])
+    strand.add_child(Strand[vm3_s.vm.nics.first.id])
+
+    current_frame = strand.stack.first
+    current_frame["vms"] = [vm1_s.id, vm2_s.id, vm3_s.id]
+    current_frame["subnets"] = [subnet1_s.id, subnet2_s.id]
+    current_frame["project_id"] = project.id
+    strand.modified!(:stack)
+    strand.save_changes
+
+    hop_wait_children_created
+  end
```

Comment:
Is there a way to make some of this more terse? Two common ways to do this is:

1. a method.
2. But if you need stuff in lexical scope, which may be the case here, another way would be a  lambda or proc

e.g.

```ruby
do_something = ->(zero_or_more_args) do
end


do_something.call(1)
do_something.call(2)
do_something.call(3)
```

This is called the "stabby lambda" operator. You can also write `lambda`.

You may also find `proc` useful. There is a difference in how they handle control flow words like `return`, e.g. I suspect in this case you want `lambda`/`->`